lib/galaxy/tools/actions/__init__.py

```diff
@@ -236,7 +236,7 @@ def execute(self, tool, trans, incoming={}, return_job=False, set_output_hid=Tru
                 incoming[ "%s|__identifier__" % name ] = identifier
 
             for tag in data.tags:
-                if tag.user_tname == 'name':
+                if tag.user_tname == 'name' and tag.value not in [t.value for t in preserved_tags]:
```

Comment:
As of the issue @mvdbeek reported it actually shows the tag only once in the tag ui (when expanded) but it shows it twice on the dataset (without expanding it)